core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/datastore/request/CodeSystemResourceRequest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.datastore.request;
+
+import com.b2international.snowowl.core.ServiceProvider;
+import com.b2international.snowowl.core.domain.BranchContext;
+import com.b2international.snowowl.core.events.DelegatingRequest;
+import com.b2international.snowowl.core.events.Request;
+import com.b2international.snowowl.core.request.SearchResourceRequest;
+import com.b2international.snowowl.core.uri.CodeSystemURI;
+import com.b2international.snowowl.datastore.CodeSystemEntry;
+import com.b2international.snowowl.datastore.CodeSystemVersionEntry;
+import com.b2international.snowowl.terminologyregistry.core.request.CodeSystemRequests;
+import com.b2international.snowowl.terminologyregistry.core.request.CodeSystemVersionSearchRequestBuilder;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * @since 7.5
+ */
+public final class CodeSystemResourceRequest<R> extends DelegatingRequest<ServiceProvider, BranchContext, R> {
+
+	private static final long serialVersionUID = 1L;
+	
+	@JsonProperty
+	private final CodeSystemURI uri;
+	
+	private transient CodeSystemEntry codeSystem;
+	
+	private transient String branchPath;
+	
+	CodeSystemResourceRequest(String codeSystemUri, Request<BranchContext, R> next) {
+		super(next);
+		this.uri = new CodeSystemURI(codeSystemUri);
+	}
+
+	@Override
+	public R execute(ServiceProvider context) {
+		return new RepositoryRequest<R>(getRepositoryId(context),
+			new BranchRequest<R>(getBranchPath(context),
+				next()
+			)
+		).execute(context);
+	}
+
+	public CodeSystemEntry getCodeSystem(ServiceProvider context) {
+		if (codeSystem == null) {
+			codeSystem = CodeSystemRequests.getCodeSystem(context, uri.getCodeSystem());
+		}
+		return codeSystem;
+	}
+	
+	public String getRepositoryId(ServiceProvider context) {
+		return getCodeSystem(context).getRepositoryUuid();
+	}
+	
+	public String getBranchPath(ServiceProvider context) {
+		if (branchPath == null) {
+			CodeSystemVersionSearchRequestBuilder versionSearch = CodeSystemRequests.prepareSearchCodeSystemVersion()
+				.one()
+				.filterByCodeSystemShortName(codeSystem.getShortName());
```

Comment:
The code system version search request is only used in the `else` block below; move the local variable declaration inside the block.